src/ip_resolver.cpp

```diff
@@ -20,6 +20,73 @@
 
 #include "ip_resolver.hpp"
 
+int zmq::ip_addr_t::family () const
+{
+    return generic.sa_family;
+}
+
+bool zmq::ip_addr_t::is_multicast () const
+{
+    if (family () == AF_INET) {
+        uint32_t addr = ntohl (ipv4.sin_addr.s_addr);
+
+        //  IPv4 Multicast: address MSBs are 1110
+        //  Range: 224.0.0.0 - 239.255.255.255
+        return (addr & 0xf0000000) == 0xe0000000;
```

Comment:
the standard library has functions and macros for this:

IN_IS_ADDR_MULTICAST
IN6_IS_ADDR_MULTICAST

If they work on Windows, perhaps it's best to use them rather than re-implement?

Reply:
I considered it but I didn't know how portable it was and given that it was a one-liner I didn't risk it. I'll try the macros and we'll see if the buildbot is happy I suppose.

Reply:
We use another one of those in CZMQ which builds more or less on the same platforms, so that's why I mentioned it:

https://github.com/zeromq/czmq/blob/master/src/zbeacon.c#L192